### PR TITLE
faq: update dm docs and add 4 FAQs

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added, or deleted? (Required)

Update dm docs and add 4 FAQs.

The changes in `zh/dmctl-introduction.md` have been partly aligned to `en/dmctl-introduction.md` in https://github.com/pingcap/docs-dm/pull/560. Confirmed with @lance6716 that he will align the rest changes when DM v2.0.2 is about to release. 

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-dm/pull/551
- Other reference link(s): https://github.com/pingcap/docs-dm/pull/560

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: en/faq.md [hunk 1]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
 You need to check whether the [safe mode](glossary.md#safe-mode) is automatically enabled for the task. If the task is automatically resumed after an error, or if there is high availability scheduling, then the safe mode is enabled because it is within 5 minutes after the task is started or resumed.
 
 You can check the DM-worker log file and search for a line containing `change count`. If the `new count` in the line is not zero, the safe mode is enabled. To find out why it is enabled, check when it happens and if any errors are reported before.
+
+## In DM v2.0, why does the full import task fail if DM restarts during the task?
+
+In DM v2.0.1 and lower versions, if DM restarts before the full import completes, the bindings between upstream data sources and DM-worker nodes might change. For example, it is possible that the intermediate data of the dump unit is on DM-worker node A but the load unit is run by DM-worker node B, thus causing the operation to fail.
+
+Following are two solutions to this issue:
```
Comment (qiancai) [inline en/faq.md:208]:
```suggestion
The following are two solutions to this issue:
```

File: en/faq.md [hunk 2]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+- If the data volume is small (less than 1 TB) or the task merges sharded tables, you need to take these steps:
```
Comment (qiancai) [inline en/faq.md:210]:
```suggestion
- If the data volume is small (less than 1 TB) or the task merges sharded tables, take these steps:
```

File: en/faq.md [hunk 3]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+    1. Clean up the imported data in the downstream database.
+    2. Remove all files in the directory of exported data.
+    3. Delete the task using dmctl and run the command `start-task --remove-meta` to create a new task.
+
+    After the new task starts, it is recommended to ensure that there is no redundant DM worker node and avoid restarting or upgrading the DM cluster during this time.
+
+- If the data volume is large (more than 1 TB), you need to take these steps:
```
Comment (qiancai) [inline en/faq.md:218]:
```suggestion
- If the data volume is large (more than 1 TB), take these steps:
```

File: en/faq.md [hunk 4]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+Following are two solutions to this issue:
+
+- If the data volume is small (less than 1 TB) or the task merges sharded tables, you need to take these steps:
+
+    1. Clean up the imported data in the downstream database.
+    2. Remove all files in the directory of exported data.
+    3. Delete the task using dmctl and run the command `start-task --remove-meta` to create a new task.
+
+    After the new task starts, it is recommended to ensure that there is no redundant DM worker node and avoid restarting or upgrading the DM cluster during this time.
```
Comment (qiancai) [inline en/faq.md:216]:
```suggestion
    After the new task starts, it is recommended to ensure that there is no redundant DM worker node and avoid restarting or upgrading the DM cluster during the full import task.
```

File: en/faq.md [hunk 5]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+        - Change `task-mode` to `incremental`.
+        - Set the value of `mysql-instance.meta.pos` to the position recorded in the metadata file that the dump unit outputs.
+
+## Why does DM report the error `ERROR 1236 (HY000): The slave is connecting using CHANGE MASTER TO MASTER_AUTO_POSITION = 1, but the master has purged binary logs containing GTIDs that the slave requires.` if it restarts during an incremental task?
+
+This error indicates that the upstream binlog position recorded in the metadata file output by the dump unit has been purged during the full migration.
+
+If this issue occurs, you need to pause the task, delete all migrated data in the downstream database, and start a new task with the `--remove-meta` option.
+
+You can avoid this issue in advance by configuring in the following ways:
```
Comment (qiancai) [inline en/faq.md:233]:
```suggestion
You can avoid this issue in advance by making the following configurations:
```

File: en/faq.md [hunk 6]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+1. Increase the value of `expire_logs_days` in the upstream MySQL database to avoid wrongly purging needed binlog files before the full migration task completes. If the data volume is large, it is recommended to use dumpling and TiDB-Lightning at the same time to speed up the task.
+2. Enable the relay log feature for this task so that DM can read data from relay logs even though the binlog position is purged.
+
+## Why does the Grafana dashboard of a DM cluster displays `failed to fetch dashboard` if the cluster is deployed through TiUP v1.3.0 or v1.3.1?
```
Comment (qiancai) [inline en/faq.md:238]:
```suggestion
## Why does the Grafana dashboard of a DM cluster displays `failed to fetch dashboard` if the cluster is deployed using TiUP v1.3.0 or v1.3.1?
```

File: en/faq.md [hunk 7]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+                }
+            ]
+        }
+    ]
+}
+```
+
+In the above example, the `syncerBinlogGtid` of data source `mysql1` is inconsecutive. You can handle the data loss according to either of the following solutions:
+
+- If upstream binlogs from current time to the position recorded in the metadata of the full export task have not been purged, you can take the these steps:
```
Comment (qiancai) [inline en/faq.md:335]:
```suggestion
- If upstream binlogs from the current time to the position recorded in the metadata of the full export task have not been purged, you can take  these steps:
```

File: en/faq.md [hunk 8]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+If this issue occurs, you need to pause the task, delete all migrated data in the downstream database, and start a new task with the `--remove-meta` option.
+
+You can avoid this issue in advance by configuring in the following ways:
+
+1. Increase the value of `expire_logs_days` in the upstream MySQL database to avoid wrongly purging needed binlog files before the full migration task completes. If the data volume is large, it is recommended to use dumpling and TiDB-Lightning at the same time to speed up the task.
+2. Enable the relay log feature for this task so that DM can read data from relay logs even though the binlog position is purged.
+
+## Why does the Grafana dashboard of a DM cluster displays `failed to fetch dashboard` if the cluster is deployed through TiUP v1.3.0 or v1.3.1?
+
+This is a known bug of TiUP, which is fixed in TiUP v1.3.2. Following are two solutions to this issue:
```
Comment (qiancai) [inline en/faq.md:240]:
```suggestion
This is a known bug of TiUP, which is fixed in TiUP v1.3.2. The following are two solutions to this issue:
```

File: en/faq.md [hunk 9]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+- Method one:
```
Comment (qiancai) [inline en/faq.md:242]:
```suggestion
- Solution one:
```

File: en/faq.md [hunk 10]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+    1. Upgrade TiUP to a higher version using the command `tiup update --self && tiup update dm`.
+    2. Scale in and then scale out Grafana nodes in the cluster to restart the Grafana service.
+- Method two:
```
Comment (qiancai) [inline en/faq.md:245]:
```suggestion
- Solution two:
```

File: en/faq.md [hunk 11]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+1. Increase the value of `expire_logs_days` in the upstream MySQL database to avoid wrongly purging needed binlog files before the full migration task completes. If the data volume is large, it is recommended to use dumpling and TiDB-Lightning at the same time to speed up the task.
+2. Enable the relay log feature for this task so that DM can read data from relay logs even though the binlog position is purged.
+
+## Why does the Grafana dashboard of a DM cluster displays `failed to fetch dashboard` if the cluster is deployed through TiUP v1.3.0 or v1.3.1?
+
+This is a known bug of TiUP, which is fixed in TiUP v1.3.2. Following are two solutions to this issue:
+
+- Method one:
+    1. Upgrade TiUP to a higher version using the command `tiup update --self && tiup update dm`.
```
Comment (qiancai) [inline en/faq.md:243]:
```suggestion
    1. Upgrade TiUP to a later version using the command `tiup update --self && tiup update dm`.
```

File: en/faq.md [hunk 12]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+    3. Unpack the `grafana-v4.0.3-**.tar.gz` in the offline package.
+    4. Replace the folder `deploy/grafana-$port/bin/public` with the `public` folder in `grafana-v4.0.3-**.tar.gz`.
+    5. Execute `tiup dm restart $cluster_name -R grafana` to restart the Grafana service.
+
+## In DM v2.0, why does the query result of the command `query-status` show that the Syncer checkpoint GTIDs are inconsecutive if the task has `enable-relay` and `enable-gtid` enabled at the same time?
+
+This is a known bug in DM, which is fixed in DM v2.0.2. The bug is triggered when the following two conditions are fully met at the same time:
+
+1. Parameters `enable-relay` and `enable-gtid` are set to `true` in the source configuration file.
+2. The upstream database is a **MySQL secondary database** and if you execute the command `show binlog events in '<newest-binlog>' limit 2` to query the `previous_gtids` of the database, the result is inconsecutive, such as the following example:
```
Comment (qiancai) [inline en/faq.md:257]:
```suggestion
2. The upstream database is a **MySQL secondary database**. If you execute the command `show binlog events in '<newest-binlog>' limit 2` to query the `previous_gtids` of the database, the result is inconsecutive, such as the following example:
```

File: en/faq.md [hunk 13]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+                        "binlogType": "local"
+                    }
+                }
+            ]
+        }
+    ]
+}
+```
+
+In the above example, the `syncerBinlogGtid` of data source `mysql1` is inconsecutive. You can handle the data loss according to either of the following solutions:
```
Comment (qiancai) [inline en/faq.md:333]: 'either' is usually used when there are two options, but it seems that there are three solutions. 
```suggestion
In the example, the `syncerBinlogGtid` of the data source `mysql1` is inconsecutive. In this case, you can do one of the following to handle the data loss:
```

File: en/faq.md [hunk 14]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+
+- If upstream binlogs from current time to the position recorded in the metadata of the full export task have not been purged, you can take the these steps:
+    1. Pause the current task and delete all data sources with inconsecutive GTIDs.
+    2. Set `enable-relay` to `false` in all source configuration files.
+    3. For data sources with inconsecutive GTIDs (such as `mysql1` in the above example), change the task to an incremental task and configure related `mysql-instances.meta` with metadata information of each full export task, including the `binlog-name`, `binlog-pos`, and `binlog-gtid` information.
```
Comment (qiancai) [inline en/faq.md:338]: 这句原文里的“重启任务” 好像没有出现在翻译中。

File: en/faq.md [hunk 15]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+    4. Set `syncers.safe-mode` to `true` in `task.yaml` of the incremental task.
+    5. After the incremental task replicates all missing data to the downstream, restart the task and set `safe-mode` to `false`.
+- If upstream binlogs have been purged but local relay logs remain, you can take these steps:
+    1. Pause the current task.
+    2. For data sources with inconsecutive GTIDs (such as `mysql1` in the above example), change the task to an incremental task and configure related `mysql-instances.meta` with metadata information of each full export task, including the `binlog-name`, `binlog-pos`, and `binlog-gtid` information.
```
Comment (qiancai) [inline en/faq.md:343]: 这句原文里的“重启任务” 好像没有出现在翻译中。

File: en/faq.md [hunk 16]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+    3. In the `task.yaml` of the incremental task, change the previous value of `binlog-gtid` to the previous value of `previous_gtids`. For the above example, change `1-y` to `6-y`.
+    4. Set `syncers.safe-mode` to `true` in the `task.yaml`.
+    5. After the incremental task replicates all missing data to the downstream, restart the task and set `safe-mode` to `false`.
+    6. Restart the data source and set either `enable-relay` or `enable-gtid` to `false` in the source configuration file.
+- If none of the above conditions are met or if the data volume of the task is small, you can take these steps:
```
Comment (qiancai) [inline en/faq.md:348]:
```suggestion
- If none of the above conditions is met or if the data volume of the task is small, you can take these steps:
```

**Review by qiancai — COMMENTED** [review]
In https://github.com/pingcap/docs-dm/pull/551/files, the "zh/dmctl-introduction.md” file is changed, but in this PR, there is no corresponding file. Is the translation done in other PRs?

**Comment by lance6716** [comment]
> In https://github.com/pingcap/docs-dm/pull/551/files, the "zh/dmctl-introduction.md” file is changed, but in this PR, there is no corresponding file. Is the translation done in other PRs?

As the PR description says "The changes in zh/dmctl-introduction.md have been partly aligned to en/dmctl-introduction.md in #560"

**Comment by qiancai** [comment]
> > In https://github.com/pingcap/docs-dm/pull/551/files, the "zh/dmctl-introduction.md” file is changed, but in this PR, there is no corresponding file. Is the translation done in other PRs?
> 
> As the PR description says "The changes in zh/dmctl-introduction.md have been partly aligned to en/dmctl-introduction.md in #560"

Ok. Thanks for the explanation. Then please ignore this comment.

File: en/faq.md (hunk 14 above)
Comment (CharLotteiu) [reply en/faq.md:338]: 这里中文的“重启任务”我理解就是指修改原来的 task.yaml 文件，在文件中进行各种参数配置。因为第 4 步也是在改 task.yaml 里面的一个配置值，所以应该是在第 4 步结束后，运行这个修改过后的 task.yaml 文件，开始增量复制，才算是实际意义上的“重启任务”。

麻烦 @lance6716 帮忙确认一下，如果是这样的话我在第 4 步最后加一下重启任务的描述。

File: en/faq.md (hunk 15 above)
Comment (CharLotteiu) [reply en/faq.md:343]: 同 https://github.com/pingcap/docs-dm/pull/593/files#r612299902

File: en/faq.md (hunk 14 above)
Comment (lance6716) [reply en/faq.md:338]: 第一步停止任务（突然发现line 336不是Pause而是Stop）、第三步改任务配置文件、第四步改文件并启动任务、第五步停止任务改配置再启动

中文版文档也不太行啊……等英文版合并了，我按照修改后的英文版改一下

File: en/faq.md [hunk 17]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+```
+
+In the example, the `syncerBinlogGtid` of the data source `mysql1` is inconsecutive. In this case, you can do one of the following to handle the data loss:
+
+- If upstream binlogs from the current time to the position recorded in the metadata of the full export task have not been purged, you can take these steps:
+    1. Pause the current task and delete all data sources with inconsecutive GTIDs.
+    2. Set `enable-relay` to `false` in all source configuration files.
+    3. For data sources with inconsecutive GTIDs (such as `mysql1` in the above example), change the task to an incremental task and configure related `mysql-instances.meta` with metadata information of each full export task, including the `binlog-name`, `binlog-pos`, and `binlog-gtid` information.
+    4. Set `syncers.safe-mode` to `true` in `task.yaml` of the incremental task.
+    5. After the incremental task replicates all missing data to the downstream, restart the task and set `safe-mode` to `false`.
```
Comment (CharLotteiu) [inline en/faq.md:340]:
```suggestion
    1. Stop the current task and delete all data sources with inconsecutive GTIDs.
    2. Set `enable-relay` to `false` in all source configuration files.
    3. For data sources with inconsecutive GTIDs (such as `mysql1` in the above example), change the task to an incremental task and configure related `mysql-instances.meta` with metadata information of each full export task, including the `binlog-name`, `binlog-pos`, and `binlog-gtid` information.
    4. Set `syncers.safe-mode` to `true` in `task.yaml` of the incremental task and restart the task.
    5. After the incremental task replicates all missing data to the downstream, stop the task and change `safe-mode` to `false` in the `task.yaml`.
    6. Restart the task again.
```

@lance6716 PTAL

Comment (lance6716) [reply en/faq.md:340]: lgtm

File: en/faq.md [hunk 18]
```diff
@@ -200,3 +200,154 @@ In DM v2.0, `handle-error` replaces `sql-skip`. You can use `handle-error` inste
...
+- If upstream binlogs have been purged but local relay logs remain, you can take these steps:
+    1. Pause the current task.
+    2. For data sources with inconsecutive GTIDs (such as `mysql1` in the above example), change the task to an incremental task and configure related `mysql-instances.meta` with metadata information of each full export task, including the `binlog-name`, `binlog-pos`, and `binlog-gtid` information.
+    3. In the `task.yaml` of the incremental task, change the previous value of `binlog-gtid` to the previous value of `previous_gtids`. For the above example, change `1-y` to `6-y`.
+    4. Set `syncers.safe-mode` to `true` in the `task.yaml`.
+    5. After the incremental task replicates all missing data to the downstream, restart the task and set `safe-mode` to `false`.
```
Comment (CharLotteiu) [inline en/faq.md:346]:
```suggestion
    1. Stop the current task.
    2. For data sources with inconsecutive GTIDs (such as `mysql1` in the above example), change the task to an incremental task and configure related `mysql-instances.meta` with metadata information of each full export task, including the `binlog-name`, `binlog-pos`, and `binlog-gtid` information.
    3. In the `task.yaml` of the incremental task, change the previous value of `binlog-gtid` to the previous value of `previous_gtids`. For the above example, change `1-y` to `6-y`.
    4. Set `syncers.safe-mode` to `true` in the `task.yaml` and restart the task.
    5. After the incremental task replicates all missing data to the downstream, stop the task and change `safe-mode` to `false` in the `task.yaml`.
    6. Restart the task again.
```

File: en/faq.md (hunk 14 above)
Comment (CharLotteiu) [reply en/faq.md:338]: Updated in https://github.com/pingcap/docs-dm/pull/593/commits/10972146014b1ae1e913df4f2a0d2d93ae8a82d8。 @qiancai PTAL again.

Comment (qiancai) [reply en/faq.md:338]: LGTM

**Pull request merged**